Docker/Component.php

```diff
@@ -0,0 +1,211 @@
+    public function getConfigurationFormat()
+    {
+        return $this->data['configuration_format'];
+    }
+
+    /**
+     * @return array
+     */
+    public function getImageParameters()
+    {
+        return $this->data['image_parameters'];
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasDefaultBucket()
+    {
+        return !empty($this->data['default_bucket']);
+    }
+
+    /**
+     * @param $configId
+     * @return string
+     */
+    public function getDefaultBucketName($configId)
+    {
+        return $this->data['default_bucket_stage'] . ".c-" . $this->getSanitizedComponentId() . "-" . $configId;
+    }
+
+    /**
+     * @return bool
+     */
+    public function injectEnvironment()
+    {
+        return !empty($this->data['inject_environment']);
+    }
+
+    public function forwardToken()
+    {
+        return $this->data['forward_token'];
+    }
+
+    public function forwardTokenDetails()
+    {
+        return $this->data['forward_token_details'];
+    }
+
+    public function getType()
+    {
+        return $this->data['definition']['type'];
+    }
+
+    /**
+     * @return string
+     */
+    public function getLoggerType()
+    {
+        if (!empty($this->data['logging'])) {
+            return $this->data['logging']['type'];
+        }
+        return 'standard';
+    }
+
+    /**
+     * @return array
+     */
+    public function getLoggerVerbosity()
+    {
+        if (!empty($this->data['logging'])) {
+            return $this->data['logging']['verbosity'];
+        }
+        return [];
+    }
+
+    /**
+     * @return string
+     */
+    public function getLoggerServerType()
+    {
+        if (!empty($this->data['logging'])) {
+            switch ($this->data['logging']['gelf_server_type']) {
+                case 'udp':
+                    return ServerFactory::SERVER_UDP;
+                    break;
+                case 'tcp':
+                    return ServerFactory::SERVER_TCP;
+                    break;
+                case 'http':
+                    return ServerFactory::SERVER_HTTP;
+                    break;
+                default:
+                    throw new ApplicationException(
+                        "Server type '{$this->data['logging']['gelf_server_type']}' not supported"
+                    );
+            }
+        }
+        return ServerFactory::SERVER_TCP;
+    }
+
+    /**
+     * @return string
+     */
+    public function getNetworkType()
+    {
+        return $this->networkType;
+    }
+
+    /**
+     * @return int
+     */
+    public function getCpuShares()
+    {
+        return $this->data['cpu_shares'];
+    }
+
+    /**
+     * @return string
+     */
+    public function getMemory()
+    {
+        return $this->data['memory'];
+    }
+
+    /**
+     * @return int
+     */
+    public function getProcessTimeout()
+    {
+        return (int)($this->data['process_timeout']);
+    }
+
+    public function getImageDefinition()
```

Comment:
no getDefinition mi prijde zmateny, protoze bych cekal, ze to vrati $data (tedy ze to vrati component definition)